cumulusci/robotframework/tests/salesforce/pageobjects/ObjectManagerPageObject.py

```diff
@@ -0,0 +1,144 @@
+
+import time
+from cumulusci.robotframework.utils import capture_screenshot_on_error
+from cumulusci.robotframework.pageobjects import BasePage
+from cumulusci.robotframework.pageobjects import pageobject
+from selenium.webdriver.common.keys import Keys
+
+object_manager = {"button":"//input[@title='{}']",
+				  "input":"//input[@id='{}']",
+				  "select_related":"//select[@id = '{}']",
+				  "select_related_option":"//select[@id = 'DomainEnumOrId']/option[@value='{}']",
+				  "search_result": "//tbody/tr/td/a/span[contains(text(),'{}')]",
+				  "formula_txtarea": "//textarea[@id = '{}']",
+				  "object_result": "//th/a[text()='{}']",
+				  "link-text":"//a[contains(text(),'{}')]",
+				  "button-with-text":"//button[contains(text(),'{}')]",
+				  "frame_new":"//iframe[contains(@name, '{}') or contains(@title, '{}')]"
+				  }
+
+
+@pageobject("Custom", "ObjectManager")
+class ObjectManagerPage(BasePage):
+	
+	@capture_screenshot_on_error
+	def open_fields_and_relationships(self, object_name):
+		"""To go to object manager page for a specific object"""
+		
+		url_template = "{root}/lightning/setup/ObjectManager/home"
+		url = url_template.format(root=self.cumulusci.org.lightning_base_url, object=object_name)
+		self.selenium.go_to(url)
+		search_button = object_manager["input"].format("globalQuickfind")
+		self.salesforce.wait_until_loading_is_complete()
+		self.selenium.wait_until_page_contains_element(search_button)
+		self.selenium.get_webelement(search_button).send_keys(object_name)
+		self.selenium.get_webelement(search_button).send_keys(Keys.ENTER)
+		object = object_manager["object_result"].format(object_name)
+		leftnavoption = object_manager['link-text'].format("Fields & Relationships")
+		self.selenium.wait_until_page_contains_element(object)
+		self.selenium.click_element(object)
+		self.selenium.wait_until_location_contains("Details/view", timeout=90)
+		self.selenium.click_element(leftnavoption)
+		
+	@capture_screenshot_on_error
+	def create_currency_field(self,field_name):
+		"""Creates a currency field by taking in the field name"""
+		currency_locator=object_manager['input'].format("dtypeC")
+		next_button=object_manager['button'].format("Next")
+		save_button=object_manager['button'].format("Save")
```

Comment:
I see this sort of code repeated in every method. It might be better to just have a "next" and "save" locator, so that you can skip these lines and just do `self.selenium.click_button(object_manager['next'])`. Or, create a custom keyword called `click_action_button` that takes a label and does the lookup (eg: `self.click_action_button('Save')`). 

While that only saves a couple lines of code, when you multiply that times 20+ keywords for all of the different field types, it starts to add up.

Reply:
Moved all the common locators and declared above for common reusability